app/views.py

```diff
@@ -142,6 +142,34 @@ def login():
         return redirect(url_for('dashboard'))
     return render_template('login.html', title=' | Log In', form=form)
 
+@app.route('/deactivate', methods=['GET','POST'])
+def deactivate():
+    form = DeactivateForm()
+    if form.validate_on_submit():
+        user = User.query.filter_by(email=form.email.data).first()
+        if user is None:
+            flash('Invalid email. Please try again')
+            return redirect(url_for('deactivate'))
+        return redirect(url_for('deactivate'))
+    return render_template('deactivate.html', title=' | Deactivate Account', form=form)
+
+@app.route('/deactivate/<token>')
+@login_required
+def deactivate_email(token):
+    try:
+        email = deactivate_token(token)
+    except:
+        flash('The confirmation link is invalid or has expired.', 'danger')
+        return redirect(url_for('deactivate'))
+    user = User.query.filter_by(email=email).first_or_404()
+    if user.confirmed:
+        flash('Account already confirmed. Please login.', 'success')
+    else:
+        db.session.delete(user)
+        db.session.commit()
+        flash('You have deactivated your account successfully!', 'success')
+    return redirect(url_for('home'))
```

Comment:
If we follow the REST API, we might want to have this be a DELETE request to a user URI instead. Similar for reset password being a PUT request to the user as well.